test/FsAutoComplete.Tests.Lsp/InlayHintTests.fs

```diff
@@ -108,12 +157,321 @@ let tests state =
         """
            []
 
-      testCaseAsync "type hints are truncated to 30 characters"
-      <| InlayHints.check
-           server
-           """
-        $0let t = Some (Some (Some (Some (Some (Some (Some (Some (Some (Some (Some (Some (Some (Some (Some ()))))))))))))))$0
-        """
-           [ "unit option option option option option option option option option option option option option option option",
-             (0, 5),
-             InlayHintKind.Type ] ])
+      testCaseAsync "show: param & variable have different names" <|
+        InlayHints.checkRange server
+          """
+          let f beta = ()
+          let alpha = 42
+
+          $0f $0alpha$0
+          """
+          [ param "beta" ]
+
+      testCaseAsync "hide: param & variable have same name" <|
+        InlayHints.checkRange server
+          """
+          let f alpha = ()
+          let alpha = 42
+
+          $0f alpha$0
+          """
+          [  ]
+      testCaseAsync "hide: variable prefix of param" <|
+        InlayHints.checkRange server
+          """
+          let f rangeCoveringExpr = ()
+          let range = 2
+
+          $0f range$0
+          """
+          [  ]
+      testCaseAsync "hide: variable postfix of param" <|
+        InlayHints.checkRange server
+          """
+          let f exactRange = ()
+          let range = 2
+
+          $0f range$0
+          """
+          [  ]
+      //todo: or hide?
+      testCaseAsync "show: variable infix of param" <|
```

Comment:
I think these three cases are fine to keep showing.  word boundaries are a good boundary, the rest we can't make a reasonable rule for IMO.